storage/acl.go

```diff
@@ -77,8 +77,13 @@ type ACLHandle struct {
 
 // Delete permanently deletes the ACL entry for the given entity.
 func (a *ACLHandle) Delete(ctx context.Context, entity ACLEntity) (err error) {
-	ctx = trace.StartSpan(ctx, "cloud.google.com/go/storage.ACL.Delete")
-	defer func() { trace.EndSpan(ctx, err) }()
+	if isOTelTracingDevEnabled() {
```

Comment:
Can we handle this check in startSpan rather than at each individual callsite? It would reduce a lot of boilerplate.

Reply:
Absolutely, this makes much more sense and is easier for cleanup. Updated PTAL, thanks!